test/testdrive/dates-times.td

```diff
@@ -109,3 +165,13 @@ $ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
 
 > SELECT '1989-06-01 10:10:10.1234564+04:00'::timestamptz::text
 "1989-06-01 06:10:10.123456+00"
+
+# Timestamp columns with precisions
+> DROP TABLE IF EXISTS temporal_types;
+
+> CREATE TABLE temporal_types (timestamp_col TIMESTAMP, timestamptz_col TIMESTAMPTZ, timestamp_prec_col TIMESTAMP(3), timestamptz_prec_col TIMESTAMPTZ(1));
+
+> INSERT INTO temporal_types VALUES ('2010-10-10 10:10:10.123456789+00','2010-10-10 10:10:10.123456789+00', '2010-10-10 10:10:10.123456789+00','2010-10-10 10:10:10.123456789+00');
+
+> SELECT * FROM temporal_types;
+"2010-10-10 10:10:10.123457" "2010-10-10 10:10:10.123457 UTC" "2010-10-10 10:10:10.123" "2010-10-10 10:10:10.100 UTC"
```

Comment:
@def- I added this from the platform checks you had added. These are passing, but not exactly sure why the nightlies are failing https://buildkite.com/materialize/nightlies/builds/3925#018a76ec-b026-48c0-9259-f5246caa97a1/174-1387

Reply:
Oh right, that is my mistake. We can't start up old Mz versions with columns with timestamp precision. I'll provide a fix.

Reply:
Makes sense, ty.